crates/typst-syntax/src/package.rs

```diff
@@ -0,0 +1,267 @@
+//! Package manifest parsing.
+
+use std::fmt::{self, Debug, Display, Formatter};
+use std::str::FromStr;
+
+use ecow::{eco_format, EcoString};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use unscanny::Scanner;
+
+use crate::is_ident;
+
+/// A parsed package manifest.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct PackageManifest {
+    /// Details about the package itself.
+    pub package: PackageInfo,
+    /// Details about the template, if the package is one.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub template: Option<TemplateInfo>,
+}
+
+/// The `[template]` key in the manifest.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct TemplateInfo {
+    /// The path of the starting point within the package.
+    pub path: EcoString,
+    /// The path of the entrypoint relative to the starting point's `path`.
+    pub entrypoint: EcoString,
+}
+
+/// The `[package]` key in the manifest.
+///
+/// More fields are specified, but they are not relevant to the compiler.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct PackageInfo {
+    /// The name of the package within its namespace.
+    pub name: EcoString,
+    /// The package's version.
+    pub version: PackageVersion,
+    /// The path of the entrypoint into the package.
+    pub entrypoint: EcoString,
+    /// The minimum required compiler version for the package.
+    pub compiler: Option<PackageVersion>,
+}
+
+impl PackageManifest {
+    /// Ensure that this manifest is indeed for the specified package.
+    pub fn validate(&self, spec: &PackageSpec) -> Result<(), EcoString> {
+        if self.package.name != spec.name {
+            return Err(eco_format!(
+                "package manifest contains mismatched name `{}`",
+                self.package.name
+            ));
+        }
+
+        if self.package.version != spec.version {
+            return Err(eco_format!(
+                "package manifest contains mismatched version {}",
+                self.package.version
+            ));
+        }
+
+        if let Some(compiler) = self.package.compiler {
```

Comment:
I suggest to rename it for better understanding:
```suggestion
        if let Some(required) = self.package.compiler {
```